src/TextInputWithIcon/TextInputWithIcon.tsx

```diff
@@ -0,0 +1,81 @@
+import React from 'react'
+import styled, { css } from 'styled-components'
+import { IconType } from 'react-icons/src'
+
+import { TextInput, TextInputProps, OmitOnChangeHTMLInputAttributes } from '../TextInput/TextInput'
+import { matchThemeProp, InputProps, defaultStyledProps } from '../utils/styled'
+import { Icon } from '../Icon'
+
+export interface TextInputWithIconProps extends TextInputProps {
+  icon: React.ReactElement<null, IconType>
+  iconLeft?: boolean
```

Comment:
I'd call this `position: 'left' | 'right'` with a default value in defaultProps

Reply:
Yeah, thats better.

Reply:
> Yeah, thats better. although i dont know.. 'right' would be meaningless

why?